include/string.h

```diff
@@ -86,8 +85,14 @@ char *strerror (int);
 char *strtok_r (char *__restrict, const char *__restrict, char **__restrict);
 int strerror_r (int, char *, size_t);
 char *stpcpy(char *__restrict, const char *__restrict);
-char *stpncpy(char *__restrict, const char *__restrict, size_t);
-size_t strnlen (const char *, size_t);
+// Copies at most n characters from the string pointed to by src, including the terminating null byte ('\0'),
+// to the array pointed to by dest.
+char *stpncpy(char *__restrict d : itype(__restrict _Nt_array_ptr<char>) count(n),
+              const char *restrict s : itype(__restrict _Nt_array_ptr<const char>) count(n),
+              size_t n)
+  : itype(_Nt_array_ptr<char>) count(n);
+// Returns the number of bytes in the string pointed to by s, excluding the terminating null byte ('\0').
+size_t strnlen (const char * : itype(_Nt_array_ptr<const char>) count(n), size_t n);
 char *strdup (const char *);
 char *strndup (const char *, size_t);
```

Comment:
I see that several `strn*` functions have been left uncoverted. Could you please either convert them in this PR or add a comment saying why these are left out and if we have plans to convert them?

Reply:
I am converting the strn* functions and taking note of the functions that we can not convert.